consensus/src/processes/window.rs

```diff
@@ -686,4 +711,24 @@ impl BoundedSizeBlockHeap {
         self.binary_heap.push(r_sortable_block);
         true
     }
+
+    // This method is intended to be used to merge the ancestor heap with the current heap.
+    // It is expected that any ancestor heap will have lower blue work blocks than the current heap.
+    fn merge_ancestor_heap(&mut self, ancestor_heap: &mut BlockWindowHeap) {
+        // because popping has O(log(n)) complexity, we pop the excess amount first, then append the two.
+        let overflow_amount = (self.len() + ancestor_heap.len()) - self.size_bound;
```

Comment:
`self.size_bound` can be greater than `self.len() + ancestor_heap.len()` here so probably better to do:

```
let overflow_amount = (self.len() + ancestor_heap.len()).saturating_sub(self.size_bound);
```

Reply:
Was under the impression that `self.size_bound` will never be larger then the then merging with the ancestor heap, plus what was already added to the current heap.. but I am now thinking that this might be a condition if we are close to origin, or genesis. but I guess there is no harm in saturating as it will cover this potential logic and shouldn't do any harm otherwise.